build: update dependencies and repair CI lint/build/test

jcschaff opened the pull request:
**What new features does this PR implement?**
completes work in nx-updates branch from @bilalshaikh42 to update dependencies, update to Angular 15, Typescript 18, and fix linting and unit tests

**What bugs does this PR fix?**
large number of the 66 testable modules had unit test failures.  This PR will attempt to fix all broken unit tests and prepare for new development.

**Does this PR introduce any additional changes?**
merged ux and service modules from simulation-project-utils (to solve a circular dependency).

**How have you tested this PR?**
No new tests yet, but attempting to fix all failing unit tests in the repo.

**Additional information**
The goal is to make this project pass CI and deploy with initially small or no changes to the UI or backend - but rather pave way for new development.  Some simplification of the UI is planned for this PR, but it will be modest changes.

bilalshaikh42 committed:
chore: build update dependencies and migrate apps

bilalshaikh42 committed:
refactor: fix sccs import

bilalshaikh42 committed:
refactor: update styling to latest

angular material has been updated with many changes to the sass functions and mixins. Our theme system has drifted overtime and become rather non standard, so these updates were not caught by the auto migrations

Manually refactored and updated to the latest versions. This is a good target for refactoring and simplifying.

bilalshaikh42 committed:
refactor: remove resize polyfill

typescript is claiming that the condition will never evaluate to false, implying that the resize observer api is now part of the standard window. Apparently, we no longer need the polyfill?

bilalshaikh42 committed:
fix: add string type to dates

in other parts of the code, we check to see if the type of this field is a string

bilalshaikh42 committed:
fix: update http import

httpservice is now imported from @nestjs/axios. These files are auto generated, so we will need to update the generator tool to work for nest v9 and later

jcschaff committed:
fix: nx lint command for combine-api

jcschaff committed:
chore(deps): upgrade eslint, remove flex-layout, resolve nestjs/bullmq

jcschaff committed:
fix: put back new version of flex-layout

jcschaff committed:
fix: move shared ui lib forms/paging-form to simulation-project-utils module

jcschaff committed:
fix: merged simulation-project-utils ui/service, linting fixes

jcschaff committed:
fix: complete refactoring of simulation-project-utils module

jcschaff committed:
fix: complete port to new Material Design components, remove legacy

jcschaff committed:
fix: configuration changes to fix more unit tests

jcschaff committed:
fix: to @nestjs/bullmq , removed storyboook, resolved peer dependencies

bilalshaikh42 reviewed (approved):
Looking great! Thanks for replacing the custom BullMQ, was not looking forward to that

jcschaff committed:
fix: fix docker build, upgrade workspace to nx 15.9.2

jcschaff committed:
fix: update CI scripts to node 18

jcschaff committed:
fix: update nestjs/schematics to fix dependency conflict

jcschaff committed:
fix: ci change for npm cache policy to fix old cached dependencies

jcschaff committed:
fix: consider order dependency of ci steps for assuring node 18

jcschaff committed:
fix: another order dependency in ci

jcschaff committed:
fix: ci now references same commit for child workflows, should work now

jcschaff committed:
fix: roll back typescript to 4.9.5

jcschaff committed:
fix: remove codelyzer based on tslint, now we use typescript-eslint

jcschaff committed:
fix: restore caching for ci npm install

jcschaff committed:
fix: update to setup-python version 4 to install poetry also

jcschaff committed:
fix: explicitly install poetry on Github workflow runners

jcschaff committed:
fix: port to @nestjs/bullmq

jcschaff committed:
fix: update imports from combined simulation-project-utils module

jcschaff committed:
fix: linting issues identified during build phase

jcschaff committed:
fix: mat-chip-list and mat-icon material design components upgrade

jcschaff committed:
chore: update class-validator, json5, mongoose, passport for security

jcschaff committed:
fix: eager load of simulation-project-utils module, had direct reference

jcschaff committed:
fix: visualization unit test and rollback mongoose upgrade

jcschaff committed:
fix: plotly and vega simulation-viz unit tests with NgxResizeObserver

jcschaff committed:
fix:  module load problem, Plotly now loads via CDN

jcschaff committed:
fix: combine-api CI test fix, and added nx run combine-api:test_offline

switched test script from pipenv to poetry
added 'nx run combine-api:test_offline' for testing local python code within Docker container

jcschaff committed:
fix: run build/lint/test CI action sequentially to understand failure

jcschaff committed:
fix: upgraded jest.spyOn usage to fix unit test

jcschaff committed:
fix: apply correct linting rules to angular component templates

jcschaff committed:
fix: explicit jest imports for tests when ng-mock also present

jcschaff committed:
fix: typescript errors in tests, with strict eslint rules

jcschaff committed:
chore: merge create-project.module, use ng-mocks to fix paging-form test

jcschaff committed:
fix: use https for sed-ml specifications link to fix Sonar CI alert

jcschaff commented:
All CI actions pass - node, angular, python and most dependencies are up to date.
